tools/src/main/java/com/nedap/archie/archetypevalidator/validations/SpecializedDefinitionValidation.java

```diff
@@ -153,6 +160,54 @@ private void validateConformsTo(CObject cObject, CObject parentCObject) {
 
     }
 
+    private ConformanceCheckResult childNodesConformToParent(CObject childCObject, CObject parentCObject, ConformanceCheckResult conformanceCheckResult) {
+        if (!conformanceCheckResult.getErrorType().equals(ErrorType.VSONCO)) {
+            return conformanceCheckResult;
+        }
+
+        // Edge case: if sum of all occurrences of all redefined object nodes conforms to occurrences of parent object node
+        // it is valid to exclude the parent object node
+        MultiplicityInterval parentNodeOccurrences = parentCObject.effectiveOccurrences(combinedModels::referenceModelPropMultiplicity);
+        List<CObject> allRedefinedNodes = childCObject.getParent().getChildren().stream().filter(
+                child -> child.nodeIdConformsTo(parentCObject)
+                        && !child.getNodeId().equals(parentCObject.getNodeId())
+        ).collect(Collectors.toList()); // All child nodes in the child archetype
+        allRedefinedNodes.addAll(((CAttribute) flatParent.itemAtPath(parentCObject.getParent().getPath())).getChildren().stream().filter(
```

Comment:
I think the following case still does not work correctly. It is an edge case, but still:

parent:

```
[id5] occurrences matches {2..5}
```
specialized:

```
[id5.1] occurrences matches {0..3}
[id5.2] occurrences matches {2}
```

specialized of specialized:
```
[id5.1] occurrences matches {0}
[id5.0.1] occurrences matches {0..3}
```

To fix, see my previous comment - either move this so it checks on the flattened archetype against the flat parent, or implement part of the flattener logic here to check.

Reply:
oh this is even more complex - it also matters if you exclude the parent node or not (so, [id5] occurrences matches {0} or not). Because if you do, the occurrences has to replace that of the parent node. If you do not exclude it, the parent node is still there, and the sum is all of the sibling nodes including the parent node itself.

That however, makes the entire thing very tricky, and nearly impossible to specialize without removing the parent node  unless you have upperUnbounded true in the original node. Perhaps just ignore that?

Reply:
ok, maybe ignore my example above? Still implementing the flattening logic here would be nice, but perhaps less necessary than I thought?